<a href="https://colab.research.google.com/github/ilovedata24/Homework-1/blob/main/Copy_of_hw_1_1_3_%D0%9E%D1%81%D0%BD%D0%BE%D0%B2%D0%B8_%D1%80%D0%BE%D0%B1%D0%BE%D1%82%D0%B8_%D0%B7_%D1%80%D1%8F%D0%B4%D0%BA%D0%B0%D0%BC%D0%B8_%D1%82%D0%B0_%D0%BC%D0%B0%D1%82%D0%B5%D0%BC%D0%B0%D1%82%D0%B8%D1%87%D0%BD%D1%96_%D0%BE%D0%BF%D0%B5%D1%80%D0%B0%D1%86%D1%96%D1%97_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [34]:
a = 101
b = 210
c = 111
d = 402
(((a + b) * c)) / d

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [35]:
dilene = 11283
dilnyk = 89
ostatok = dilene % dilnyk
print("Остача від ділення числа", dilene, "на", dilnyk, "дорівнює:", ostatok)

Остача від ділення числа 11283 на 89 дорівнює: 69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [36]:
dilene = 11283
dilnyk = 89
chastka = dilene // dilnyk
print (chastka)

126


4. Запишіть у змінну `name` ваше імʼя.

In [37]:
name = "Nataliia"
print(name)

Nataliia


5. Запишіть у змінну `surname` ваше прізвище.

In [38]:
surname = "Ishchenko"
print(surname)

Ishchenko


6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [39]:
name_surname = name + " " + surname
print(name_surname)

Nataliia Ishchenko


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [40]:
name_surname_upper = name_surname.upper()
print(name_surname_upper)

NATALIIA ISHCHENKO


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [41]:
third_letter = name_surname[2]
print("Третя літера:", third_letter)

Третя літера: t


Очікувана відповідь (тільки з вашими даними): 'n'

\9. Виведіть першу та останню літеру вашого прізвища.

In [42]:
surname = "Ishchenko"
first_letter = surname[0]
last_letter = surname[-1]
print("Перша літера прізвища:", first_letter)
print("Остання літера прізвища:", last_letter)

Перша літера прізвища: I
Остання літера прізвища: o


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [43]:
surname = "Ishchenko"
surname_length = len(surname)
print("Довжина прізвища:", surname_length)

Довжина прізвища: 9


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [44]:
capital = "Antananarivo"
a_count = capital.lower().count('a')
print("Кількість літер 'a' у назві столиці:", a_count)

Кількість літер 'a' у назві столиці: 4


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [45]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
starts_with_the = my_string.startswith("The")
ends_with_dog = my_string.endswith("dog!")
print("Починається з 'The':", starts_with_the)
print("Закінчується на 'dog!':", ends_with_dog)

Починається з 'The': True
Закінчується на 'dog!': False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [47]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
words = my_string.split()
joined_string = "_".join(words)
print(joined_string)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`